src/assets/scripts/client/canvas/CanvasController.js

```diff
@@ -1115,16 +1094,23 @@ k
             }
```

Comment:
can you add some `TODO`s to the two `if` blocks above?  Those two need to be looked at and/or refactored to use methods in the input controller.

Actually, didn't we get rid of all the `prop` references?